src/app/projects/gard/stegnography/stegno_logic.ts

```diff
@@ -0,0 +1,79 @@
+export const loadImage = (file: File): Promise<HTMLImageElement> => {
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = (e) => {
+            const img = new Image();
+            img.onload = () => resolve(img);
+            img.onerror = reject;
+            if (e.target) img.src = e.target.result as string;
+        };
+        reader.onerror = reject;
+        reader.readAsDataURL(file);
+    });
+};
+
+
+async function hashPassword(password: string): Promise<Uint8Array> {
+  const encoder = new TextEncoder();
+  const data = encoder.encode(password);
+  const hashBuffer = await crypto.subtle.digest('SHA-256', data);
+  return new Uint8Array(hashBuffer);
+}
+
+
+function xorEnc(payload: string, key: Uint8Array): string {
+  
+  const payloadBinary = payload.split('').map((char => char.charCodeAt(0)));
+  const encryptedPayload = payloadBinary.map((charcode, i) => charcode ^ key[i % key.length]);
+  return encryptedPayload.map(byte => String.fromCharCode(byte)).join('');
+}
+
+
+
+
+export async function embedPasswordInImageData(
+  imgData: ImageData, 
+  payload: string, 
+  securityPassword: string
+): Promise<ImageData> {
+  const hashedPassword = await hashPassword(securityPassword);
+  const encryptedPayload = xorEnc(payload, hashedPassword);
+  
+  const binaryPayload = encryptedPayload
+    .split('')
+    .map(char => char.charCodeAt(0).toString(2).padStart(8, '0'))
+    .join('') + '1111111111111110'; 
```

Comment:
This concatenation is redundant. Instead of comparing this arbitrary 16-bit string, it would be more secure to compare the SHA 256 string from the hashed password of the user to verify the decryption.